### DISCUSSION TITLE
can we disable wagmi?

### DISCUSSION THREAD

**Discussion by agamtribe3** [wevm/wagmi#2908] [Q&A]
first of all, i will really appreciate for all of your help!

I build an app that need a realtime data, so i use `watch: true` for `useContractReads`, but it will burn my provider. even i had 5 provider already. below is my `configureChains` setting.

> const { chains, publicClient } = configureChains(
>   CHAINS,
>   [
>     w3mProvider({ projectId }),
>     jsonRpcProvider({
>       rpc: () => ({
>         http: quickNodeProviderUrl
>       })
>     }),
>     alchemyProvider({ apiKey: alchemyProjectId }),
>     infuraProvider({ apiKey: infuraProjectId }),
>     publicProvider()
>   ],
>   { pollingInterval: 8000 }
> )

even i already set the alchemy to number 3, my alchemy still burning. i wanna try to disable the provider call if the user idle, is there any suggestion? or maybe better solution. thanks

**Comment by StanleyDing** [comment]
Perhaps try setting [enabled props](https://wagmi.sh/react/hooks/useContractReads#enabled-optional) to false when user is idle.

```ts
const { data, isError, isLoading } = useContractReads({
    contracts: [
      {
        // ...
      }
    ],
    enabled: false,
  })
```
